Description: "LogLevel SILENT" compatibility
 "LogLevel SILENT" (-qq) was introduced in Debian openssh 1:3.0.1p1-1 to
 match the behaviour of non-free SSH, in which -q does not suppress fatal
 errors.  However, this was unintentionally broken in 1:4.6p1-2 and nobody
 complained, so we've dropped most of it.  The parts that remain are basic
 configuration file compatibility, and an adjustment to "Pseudo-terminal
 will not be allocated ..." which should be split out into a separate patch.
Author: Jonathan David Amery <jdamery@ysolde.ucam.org>
Author: Matthew Vernon <matthew@debian.org>
Author: Colin Watson <cjwatson@debian.org>
Last-Update: 2010-03-31

Index: b/log.c
===================================================================
--- a/log.c
+++ b/log.c
@@ -92,6 +92,7 @@
 	LogLevel val;
 } log_levels[] =
 {
+	{ "SILENT",	SYSLOG_LEVEL_QUIET }, /* compatibility */
 	{ "QUIET",	SYSLOG_LEVEL_QUIET },
 	{ "FATAL",	SYSLOG_LEVEL_FATAL },
 	{ "ERROR",	SYSLOG_LEVEL_ERROR },
Index: b/ssh.c
===================================================================
--- a/ssh.c
+++ b/ssh.c
@@ -680,7 +680,7 @@
 	/* Do not allocate a tty if stdin is not a tty. */
 	if ((!isatty(fileno(stdin)) || stdin_null_flag) &&
 	    options.request_tty != REQUEST_TTY_FORCE) {
-		if (tty_flag)
+		if (tty_flag && options.log_level != SYSLOG_LEVEL_QUIET)
 			logit("Pseudo-terminal will not be allocated because "
 			    "stdin is not a terminal.");
 		tty_flag = 0;
